packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,17 +217,22 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
-    removeUser: monitor(() => {
-      user = {}
-    }),
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor((key, property) => userContextManager.setContextProperty(key, sanitizeUser(property))),
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**❗ warning:** ‏sanitizing a user property is probably not the same as sanitizing a user.

Reply:
good point, I have removed it as ti doesn't make sense

Reply:
Sanitizing user properties seemed to be a good idea to me, what would prevent customers to do:

```ts
setUserProperty('id', 2)
```